### PR TITLE
release: harden repository health and prepare v0.2.0 candidate

### PR DESCRIPTION
**Pull request by liq22**
## Summary

Consolidates the locally validated `optimize/repo-health-10h-20260711` work into a **v0.2.0 release candidate** for the currently maintained public demo surface.

This PR includes five coherent change groups:

1. **Repository health gates**
   - `scripts.config_inspect` exits non-zero when any sanity check fails.
   - docs validation excludes local-only tooling/workspace paths.
   - focused regression tests cover both behaviors.
   - `torchmetrics` is declared with the Lightning runtime stack.
2. **Config-first CLI behavior**
   - `main.py` honors `--override pipeline=<PipelineName>`.
   - invalid pipeline values fail explicitly rather than being silently ignored.
3. **Registry and maintained-demo alignment**
   - adds the maintained `FS/classification` task and dataset mapping trace.
   - corrects the cross-system few-shot base description to `GFS`.
   - clarifies that demo 06 is a single-stage pretrain view.
4. **Generative/pretrain contract fixes**
   - aligns flow-matching training and sampling velocity semantics.
   - adds prediction-loss type, shape, and device contracts.
   - fixes masked-autoencoder class-token/masking alignment.
   - adds focused generative and pretrain contract tests.
5. **Release-candidate surface**
   - adds changelog, migration notes, supported components/combinations, known limitations, and v0.2.0 release notes.

## Release verdict

```text
READY_WITH_EXPLICIT_LIMITATIONS
```

The local handoff reports:

- release scorecard: no blockers;
- independent reviewer: `ACCEPT_WITH_RISK`;
- working tree: clean;
- source head: `f7e5d1200f5bd395b73a687201ff98a82ac16116`.

This PR prepares a release candidate only. It does **not** create the `v0.2.0` tag or GitHub Release.

## Validation evidence

Passed in the default Python environment:

```bash
python -m pytest -q \
  test/test_config_inspect_exit_code.py \
  test/test_validate_docs_scope.py \
  test/test_main_pipeline_selection.py
python -m scripts.validate_docs
python -m scripts.validate_configs
python -m py_compile scripts/config_inspect.py scripts/validate_docs.py main.py
```

Passed in the maintained `LQ_signal` environment:

```bash
python -m scripts.config_inspect \
  --config configs/demo/00_smoke/dummy_dg.yaml \
  --override trainer.num_epochs=1
python main.py \
  --config configs/demo/00_smoke/dummy_dg.yaml \
  --override trainer.num_epochs=1 \
  --override data.num_workers=0
python -m pytest test/ -q
```

Local matrix evidence reports:

- 7/7 maintained public demo one-epoch smoke commands passed;
- 3/3 invalid pipeline/model/task cases failed as expected;
- focused generative/pretrain contract tests passed.

Raw logs remain in the intentionally untracked local evidence directory:

```text
results/repo_optimization/20260711T003207Z/
```

No GitHub Actions status is attached to this head; the PR does not represent local evidence as CI evidence.

## Six explicit limitations

1. **Supported surface is narrow.** v0.2.0 support is limited to the seven maintained demo combinations listed in `SUPPORTED_COMBINATIONS.md`; registry discovery does not imply support for every model/task pair.
2. **Pipeline support is bounded.** `Pipeline_02_pretrain_fewshot` is supported only for the maintained single-stage pretrain demo; multi-stage `stages:` execution, `Pipeline_03`, and Pipeline 06 generative workflows remain outside the release-supported surface.
3. **Data availability is bounded.** Only `configs/demo/00_smoke/dummy_dg.yaml` is fully offline and repo-shipped; other maintained demos require external PHM-Vibench data and may require `data.data_dir` overrides.
4. **Environment evidence is bounded.** Maintained runtime validation was performed in the `LQ_signal` conda environment; a generic Python environment may lack packages such as `pytorch_lightning`, and no active GitHub Actions run is attached to this PR head.
5. **Evidence is functional, not scientific-performance evidence.** The results establish smoke, contract, checkpoint, and clean-exit behavior; they do not establish benchmark accuracy, state-of-the-art performance, efficiency gains, or cross-dataset scientific claims.
6. **Compatibility/test scope is bounded.** `python -m pytest test/ -q` is the maintained gate; broad historical tests are diagnostic, sampler compatibility must follow current runtime routing, and dataset-adapter fallback behavior remains a documented extension risk.

## Integration with current `main`

Current `main` includes PR #44. PR #44 changes only `apps/streamlit/**` and its two focused tests; PR #45 changes a disjoint file set. GitHub currently reports this PR as mergeable.

## Merge and rollback

Recommended merge method: **squash**.

Rollback method: revert the single squash-merge commit. Release tagging remains a separate maintainer decision after post-merge verification.

### CODE REVIEW

**Review by liq22 — COMMENTED** [review]
Maintainer review verdict: **ACCEPT_WITH_RISK**.

Reviewed against the v0.2.0 SOP and the supplied local handoff.

Accepted evidence and controls:
- source head is fixed at `f7e5d1200f5bd395b73a687201ff98a82ac16116`;
- GitHub reports the PR mergeable and there are no unresolved review threads;
- current `main` includes PR #44, whose file set is disjoint from PR #45;
- repository-health, CLI-selection, registry-alignment, generative/pretrain contract changes have focused regression tests;
- the maintained local gate reports `python -m pytest test/ -q`, seven maintained demo smokes, and three expected invalid-case failures;
- release docs now state exactly six explicit limitations and do not claim benchmark performance.

Accepted risks:
- no GitHub Actions status is attached to this head;
- runtime evidence is local and environment-specific (`LQ_signal`);
- support is limited to the documented maintained demo surface;
- Pipeline 03, Pipeline 06, multi-stage pretraining, broad registry combinations, and performance claims remain out of scope.

Decision applies to merging this as a **release candidate**, not to creating the final `v0.2.0` tag or GitHub Release. Recommended merge method: squash. Rollback: revert the squash commit.

**Comment by liq22** [comment]
## Post-merge release-candidate handoff

PR #45 was squash-merged into `main` as:

```text
fc83a44285d317920ea9633962e920500dd47788
```

Current verdict remains:

```text
READY_WITH_EXPLICIT_LIMITATIONS
```

No `v0.2.0` tag or GitHub Release has been created. No GitHub Actions status is attached to the merge commit, so the following post-merge local gate is mandatory before tagging:

```bash
git fetch origin --prune
git switch main
git pull --ff-only origin main
test "$(git rev-parse HEAD)" = "fc83a44285d317920ea9633962e920500dd47788"

python -m scripts.validate_docs
python -m scripts.validate_configs
python -m scripts.gen_config_atlas
git diff --exit-code docs/CONFIG_ATLAS.md

conda run -n LQ_signal python -m pytest test/ -q

conda run -n LQ_signal python main.py \
  --config configs/demo/00_smoke/dummy_dg.yaml \
  --override trainer.num_epochs=1 \
  --override data.num_workers=0
```

Then re-run the seven maintained public demo smoke commands and the three expected invalid pipeline/model/task cases from the external evidence pack. Record the raw logs and obtain a post-merge independent reviewer verdict.

Tagging gate:

- all commands above pass on merged `main`;
- supported-demo matrix remains consistent with `SUPPORTED_COMBINATIONS.md`;
- P0 = 0 and P1 = 0;
- reviewer verdict is `ACCEPT` or `ACCEPT_WITH_RISK` with the six published limitations unchanged.

Rollback on regression:

```bash
git switch main
git pull --ff-only origin main
git revert fc83a44285d317920ea9633962e920500dd47788
git push origin main
```

Remaining feature/audit PRs are not part of this release-candidate merge and must be updated/revalidated separately against the new `main`.
